Assignments/Assignment5.py

```diff
@@ -0,0 +1,421 @@
+    # fig.show()
+    fig.write_html(
+        file=f"plots/{pred}_mean_diff.html",
+        include_plotlyjs="cdn",
+    )
+    return file
+
+
+def main():
+    db_user = "root"
+    db_pass = input("Please enter password: ")  # pragma: allowlist secret
+    db_host = "localhost"
+    db_database = "baseball"
+    connect_string = (
+        f"mariadb+mariadbconnector://{db_user}:{db_pass}@{db_host}/{db_database}"
+    )
+
+    sql_engine = sqlalchemy.create_engine(connect_string)
+    query = """
+            SELECT *
+            FROM join_team_ba_game
+        """
+    df = pd.read_sql_query(query, sql_engine)
+    pd.set_option("display.max_columns", 20)
+    # print(df.head())
+    response = "home_team_wins"
+    preds_full = list(df.columns[:-1])
+    preds = list(
+        df.columns[6:-1]
+    )  # This is the list of preds that are most useful in predicting the wins
+
+    baseball_cor_table, cor_matrix = cor_calc_cont_cont(df, preds, response)
+
+    # I was able to add the hyper links to the plots but they seem not to click.  Im not sure why,
+    # I tested multiple file paths including the full path.
+    baseball_cor_table["Predictor 1"] = baseball_cor_table.apply(
+        lambda x: '<a href="{}">{}</a>'.format(x["pred1_plot"], x["Predictor 1"]),
+        axis=1,
+    )
+    baseball_cor_table["Predictor 2"] = baseball_cor_table.apply(
+        lambda x: '<a href="{}">{}</a>'.format(x["pred2_plot"], x["Predictor 2"]),
+        axis=1,
+    )
+    baseball_cor_table.to_html(
+        "./html_files/baseball_cor_table.html", escape=False, render_links=True
+    )
+
+    # This is a heatmap of the correlations between the predictors in the baseball dataset.
+    fig = ff.create_annotated_heatmap(cor_matrix, preds, preds, showscale=True)
+    fig.write_html(
+        file="./plots/baseball_cor_matrix.html",
+        include_plotlyjs="cdn",
+    )
+    # fig.show()
+
+    brute_force_metrics = []
+    mean_diff_metric_list = []
+
+    for p1 in preds:
+        diff_mean_df, mean_diff_metric = mean_diff_tbl(p1, response, df)
+        plot_mean_diff(diff_mean_df, p1)
+        mean_diff_metric_list.append((mean_diff_metric, p1))
+        if is_continuous(df, p1):
+            plot_contp_catr(p1, response, df)
+        else:
+            plot_catp_catr(p1, response, df)
+
+        for p2 in preds:
+            calc_matrix, brute_force_metric = brute_force(p1, p2, response, df)
+            brute_force_metrics.append((brute_force_metric, p1 + " and " + p2))
+
+            # print(calc_matrix)
+
+    sorted_BFM = sorted(brute_force_metrics, reverse=True)
+    print("Brute force Metric Sorted")
+    print(np.array(sorted_BFM), "\n")
+
+    sorted_MDM = sorted(mean_diff_metric_list, reverse=True)
+    print("Difference of Mean Metric Sorted")
+    print(np.array(sorted_MDM), "\n")
+
+    train = df[df["year"] < 2012]
+    test = df[df["year"] == 2012]
+
+    log_reg = LogisticRegression().fit(train[preds_full], train[response])
```

Comment:
you could also print out the feature importance tables here as well. log_reg.feature_importances_ should get you what you're looking for.